### PR TITLE
codecoverage/bot: Upload coverage data to Phabricator using arcanist autotarget

### PR DESCRIPTION
**Pull request by marco-c**
This is a temporary implementation until https://bugzilla.mozilla.org/show_bug.cgi?id=1487843 is resolved. The implementation won't change much after that, so most of the code will still be useful.

### CODE REVIEW

File: lib/cli_common/cli_common/phabricator.py [hunk 1]
```diff
@@ -130,15 +130,21 @@ def load_raw_diff(self, diff_id):
             diffID=diff_id,
         )
 
-    def load_revision(self, phid):
+    def load_revision(self, rev_phid=None, rev_id=None):
         '''
         Find details of a differential revision
         '''
+        assert rev_phid is not None or rev_id is not None
```
Comment (La0) [inline lib/cli_common/cli_common/phabricator.py]: Should probably be a XOR

File: src/codecoverage/bot/code_coverage_bot/phabricator.py [hunk 2]
```diff
@@ -123,6 +122,18 @@ def generate(self, report):
...
+        for rev_id, coverage in results.items():
+            logger.info('{} coverage: {}'.format(rev_id, coverage))
+
+            if not phabricator or not coverage:
+                continue
+
+            rev_data = phabricator.load_revision(rev_id=rev_id)
+            phabricator.upload_coverage_results(rev_data['fields']['diffPHID'], coverage)
+            # XXX: This is only necessary until https://bugzilla.mozilla.org/show_bug.cgi?id=1487843 is resolved.
+            phabricator.upload_lint_results(rev_data['fields']['diffPHID'], 'pass', [])
```
Comment (La0) [inline src/codecoverage/bot/code_coverage_bot/phabricator.py]: So you already had the implementation but were not using it ?

File: src/codecoverage/bot/tests/test_phabricator.py [hunk 3]
```diff
@@ -56,9 +58,51 @@ def test_upload_simple(mock_secrets, mock_phabricator, fake_hg_repo):
...
+    params = json.loads(urllib.parse.parse_qs(call.request.body)['params'][0])
+    assert params['objectPHID'] == 'PHID-DIFF-test'
+    assert params['targetKeys'] == ['arcanist.unit']
+
+    call = responses.calls[-3]
+    assert call.request.url == 'http://phabricator.test/api/harbormaster.sendmessage'
+    params = json.loads(urllib.parse.parse_qs(call.request.body)['params'][0])
+    assert params['buildTargetPHID'] == 'PHID-HMBT-test'
+    assert params['type'] == 'pass'
+    assert params['unit'] == [{'name': 'Aggregate coverage information', 'result': 'pass', 'coverage': {'file': 'NUCCCCU'}}]
```
Comment (La0) [inline src/codecoverage/bot/tests/test_phabricator.py]: :+1:

File: src/codecoverage/bot/code_coverage_bot/phabricator.py (hunk 2 above)
Comment (marco-c) [reply src/codecoverage/bot/code_coverage_bot/phabricator.py]: Yes, for a few months :smile:

**Pull request merged**